File: Low Level Design (LLD)/DesignPatterns/3. Structural/Interpreter/Cpp/main.cpp
#include <iostream>
#include "./include/Context.h"
#include "./include/VariableExpression.h"
#include "./include/NumberExpression.h"
#include "./include/AddExpression.h"

int main() {
    Context context;
    context.assign("a", 5);

    Expression* a = new VariableExpression("a");
    Expression* five = new NumberExpression(5);
    Expression* addExpr = new AddExpression(a, five);

    std::cout << "a + 5 = " << addExpr->interpret(context) << std::endl;

    delete addExpr;
    return 0;
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Interpreter/Java/Main.java
public class Main {
    public static void main(String[] args) {
        Context context = new Context();

        // Assign variables
        context.assign("a", 5);
        context.assign("b", 10);

        // a + b
        Expression a = new VariableExpression("a");
        Expression b = new VariableExpression("b");
        Expression add = new AddExpression(a, b);

        System.out.println("a + b = " + add.interpret(context)); // Output: 15
    }
}
